src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -158,16 +153,9 @@ async Task EnsureLong()
                 try
                 {
                     var finalCt = cancellationToken;
-                    if (async && Timeout > TimeSpan.Zero)
+                    if (async && Timeout > System.Threading.Timeout.InfiniteTimeSpan)
```

Comment:
Is this correct? InfiniteTimeSpan is -1 millisecond, so this will enter the condition for 0.

Reply:
You're right.

Reply:
> You're right.
... kind of...

As it stands we'd convert values of 0 and lower to `Timeout.InfiniteTimeSpan` in `NpgsqlConnector` whenever we set the value for `Timeout` so a value of 0 in the `ConnectionString` would end up as -1 here.